### PR TITLE
feat(paywalls): decode top-level state declaration map

### PR DESCRIPTION
**Pull request by MonikaMateska**
### Checklist
- [x] If applicable, unit tests
- [x] If applicable, create follow-up issues for `purchases-android` and hybrids

### Motivation
First PR in the state-driven paywalls Phase 0 stack (iOS, [PWENG-56](https://linear.app/revenuecat/issue/PWENG-56/ios-foundation-state-store-resolver-hook-forward-compat)). Adds the top-level `state` map so the SDK can decode declared state keys (type + default) that seed the presentation-session store in later PRs. Strictly additive and decode-only. 
Spec: https://github.com/RevenueCat/sdk-specs/tree/main/openspec/changes/add-paywall-component-state

### Description
Decodes the optional top-level `state` declaration on `PaywallComponentsData` into `StateDeclaration` values. Decoding is resilient: a malformed entry is dropped individually and a malformed map is ignored, so it can never fail the whole paywall. All types are `@_spi(Internal)`.

Stack: 1/5 → base `main`.

---

> [!NOTE]
> **Low Risk**
> Additive, internal SPI decoding with no runtime state-store or UI behavior yet; malformed state is dropped without failing paywall loads.
> 
> **Overview**
> Adds **decode-only** support for the optional top-level `state` map on paywall component JSON, as the first step toward state-driven paywalls.
> 
> Introduces `PaywallComponent.StateDeclaration` (wire `type` + `default` as existing `ConditionValue`, plus `normalizedDefaultValue` for double keys with integral JSON literals). `PaywallComponentsData` gains `stateDeclarations`, mapped from JSON key `state`, with resilient decoding: malformed entries are skipped, a bad map is ignored, and empty/missing/all-failed maps become `nil` so paywall decode never fails on state. Encoding round-trips the map. New unit tests cover declarations and `PaywallComponentsData` edge cases. `Package.resolved` also picks up additional SPM pins (e.g. snapshot/preview tooling).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6977334987bdc8301690c57ab93f1f241222c9b4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by MonikaMateska** [comment]
* **#6993** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-ios/6993?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6992** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-ios/6992?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6991** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-ios/6991?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6990** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-ios/6990?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6989** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-ios/6989?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-ios/6989?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: Sources/Networking/Responses/RevenueCatUI/PaywallComponentsData.swift [hunk 1]
```diff
@@ -112,6 +112,11 @@ import Foundation
     /// Exit offers configuration for this paywall.
     public var exitOffers: ExitOffers?
 
+    /// Declared paywall state keys (key → type + default), used to seed the presentation-session
+    /// state store. Declared at the root of the presentation unit; `nil` when the paywall declares
+    /// no state, which behaves identically to an empty declaration.
+    public private(set) var state: [String: PaywallComponent.StateDeclaration]?
```
Comment (alexrepty) [inline Sources/Networking/Responses/RevenueCatUI/PaywallComponentsData.swift]: _Nitpick_: I wonder if it's more self-explanatory to name this something like `stateDeclarations`, since `PaywallComponentsData` has other kinds of state that it manages, and it would make it more obvious that this is related to state observation/management.

```suggestion
    public private(set) var stateDeclarations: [String: PaywallComponent.StateDeclaration]?
```

**Review by alexrepty — APPROVED** [review]
One nitpick, and one idea: since we're decoding defensively and guarding for errors, missing values and such, should we also guard against type mismatch?

I think it's fine without these things though, so please feel free to disregard.

File: Sources/Networking/Responses/RevenueCatUI/PaywallComponentsData.swift (hunk 1 above)
Comment (MonikaMateska) [reply Sources/Networking/Responses/RevenueCatUI/PaywallComponentsData.swift]: I like the suggestion, will update it accordingly. Thanks!
